pkg/query/storeset.go

```diff
@@ -342,15 +344,24 @@ func (s *StoreSet) updateStoreStatus(store *storeRef, err error) {
 	defer s.storesStatusesMtx.Unlock()
 
 	now := time.Now()
-	s.storeStatuses[store.addr] = &StoreStatus{
-		Name:      store.addr,
-		LastError: err,
-		LastCheck: now,
-		Labels:    store.labels,
-		StoreType: store.storeType,
-		MinTime:   store.minTime,
-		MaxTime:   store.maxTime,
+
+	status := StoreStatus{Name: store.addr}
+	prev, ok := s.storeStatuses[store.addr]
+	if ok {
+		status = *prev
+	}
+
+	status.LastError = err
+	status.LastCheck = now
```

Comment:
```suggestion
	status.LastCheck = time.Now()
```
and remove extra var as it is used in one place only

Reply:
Yup ! Wished Goland catched that.